language/tools/move-package/src/resolution/resolution_graph.rs

```diff
@@ -526,8 +540,17 @@ impl ResolvingGraph {
         };
 
         for (dep_name, dep) in manifest.dependencies.iter().chain(additional_deps.iter()) {
-            Self::download_and_update_if_remote(*dep_name, dep)?;
+            if let Some(git_info) = &dep.git_info {
+                if !git_info.download_to.exists() {
+                    Self::increase_movey_download_count(
+                        movey_constants::MOVEY_URL.to_string(),
+                        git_info,
+                        build_options.skip_movey,
+                    );
+                }
+            }
 
+            Self::download_and_update_if_remote(*dep_name, dep)?;
```

Comment:
I feel like it would make more sense to include this logic in `download_and_update_if_remote`. Is there some reason not to?